### ISSUE TITLE
could not read zone configuration: Invalid status: 400 Bad request. 

### ISSUE THREAD

**Issue by zerolaser** [Venafi/terraform-provider-venafi#23]
I'm using the latest venafi provider version 0.6.3 and I was getting the following error. 

```
venafi_certificate.webserver: Creating...
  algorithm:         "" => "RSA"
  certificate:       "" => "<computed>"
  certificate_dn:    "" => "<computed>"
2019-04-11T12:03:02.140-0700 [DEBUG] plugin.terraform-provider-venafi: 2019/04/11 12:03:02 Creating certificate
  chain:             "" => "<computed>"
  common_name:       "" => "test3.nikecloud.net"
  csr_pem:           "" => "<computed>"
  ecdsa_curve:       "" => "p256"
  expiration_window: "" => "730"
  key_password:      "<sensitive>" => "<sensitive>"
  private_key_pem:   "" => "<computed>"
  rsa_bits:          "" => "2048"
2019-04-11T12:03:02.517-0700 [DEBUG] plugin.terraform-provider-venafi: 2019/04/11 12:03:02 Got 200 OK status for POST https://venafi-url:443/vedsdk/authorize/
2019-04-11T12:03:02.519-0700 [DEBUG] plugin.terraform-provider-venafi: 2019/04/11 12:03:02 Got 200 OK status for GET https://venafi-url:443/vedsdk/
2019-04-11T12:03:02.520-0700 [DEBUG] plugin.terraform-provider-venafi: 2019/04/11 12:03:02 Venafi ping sucessfull
2019-04-11T12:03:02.520-0700 [DEBUG] plugin.terraform-provider-venafi: 2019/04/11 12:03:02 Adding CN test3.nikecloud.net to SAN [] because it wasn't included.
2019-04-11T12:03:02.520-0700 [DEBUG] plugin.terraform-provider-venafi: 2019/04/11 12:03:02 Using CN test3.nikecloud.net and SAN [test3.nikecloud.net]
2019-04-11T12:03:02.520-0700 [DEBUG] plugin.terraform-provider-venafi: 2019/04/11 12:03:02 Requested SAN: [test3.nikecloud.net]
2019-04-11T12:03:02.753-0700 [DEBUG] plugin.terraform-provider-venafi: 2019/04/11 12:03:02 Making certificate request
2019-04-11T12:03:02.760-0700 [DEBUG] plugin.terraform-provider-venafi: 2019/04/11 12:03:02 Got 400 Bad request. Check the error in the response for details. status for POST https://venafi-url:443/vedsdk/certificates/checkpolicy
2019/04/11 12:03:02 [TRACE] root: eval: *terraform.EvalWriteState
2019/04/11 12:03:02 [TRACE] root: eval: *terraform.EvalApplyProvisioners
2019/04/11 12:03:02 [TRACE] root: eval: *terraform.EvalIf
2019/04/11 12:03:02 [TRACE] root: eval: *terraform.EvalWriteState
2019/04/11 12:03:02 [TRACE] root: eval: *terraform.EvalWriteDiff
2019/04/11 12:03:02 [TRACE] root: eval: *terraform.EvalApplyPost
2019/04/11 12:03:02 [ERROR] root: eval: *terraform.EvalApplyPost, err: 1 error(s) occurred:

* venafi_certificate.webserver: could not read zone configuration: Invalid status: 400 Bad request. Check the error in the response for details.
2019/04/11 12:03:02 [ERROR] root: eval: *terraform.EvalSequence, err: 1 error(s) occurred:

* venafi_certificate.webserver: could not read zone configuration: Invalid status: 400 Bad request. Check the error in the response for details.
2019/04/11 12:03:02 [TRACE] [walkApply] Exiting eval tree: venafi_certificate.webserver
2019/04/11 12:03:02 [TRACE] dag/walk: upstream errored, not walking "meta.count-boundary (count boundary fixup)"
2019/04/11 12:03:02 [TRACE] dag/walk: upstream errored, not walking "provider.venafi (close)"
2019/04/11 12:03:02 [TRACE] dag/walk: upstream errored, not walking "root"
2019/04/11 12:03:02 [TRACE] Preserving existing state lineage "57090a14-ca2a-8b26-eb4c-86d2f160cbae"
2019/04/11 12:03:02 [TRACE] Preserving existing state lineage "57090a14-ca2a-8b26-eb4c-86d2f160cbae"
2019/04/11 12:03:02 [TRACE] Preserving existing state lineage "57090a14-ca2a-8b26-eb4c-86d2f160cbae"
2019/04/11 12:03:02 [TRACE] Preserving existing state lineage "57090a14-ca2a-8b26-eb4c-86d2f160cbae"

2019/04/11 12:03:02 [DEBUG] plugin: waiting for all plugin processes to complete...
2019-04-11T12:03:02.765-0700 [DEBUG] plugin.terraform-provider-venafi: 2019/04/11 12:03:02 [ERR] plugin: plugin server: accept unix /var/folders/xn/mcw6ky311sx1nqsrj8ccbhbm0000gn/T/plugin589811456: use of closed network connection
2019-04-11T12:03:02.765-0700 [DEBUG] plugin.terraform-provider-venafi: 2019/04/11 12:03:02 [ERR] plugin: stream copy 'stderr' error: stream closed
Error: Error applying plan:

1 error(s) occurred:

* venafi_certificate.webserver: 1 error(s) occurred:

* venafi_certificate.webserver: could not read zone configuration: Invalid status: 400 Bad request. Check the error in the response for details.

Terraform does not automatically rollback in the face of errors.
Instead, your Terraform state file has been partially updated with
any resources that successfully completed. Please address the error
2019-04-11T12:03:02.767-0700 [DEBUG] plugin: plugin process exited: path=/usr/local/bin/terraform-provider-venafi
above and apply again to incrementally change your infrastructure.
```
I checked with my venafi team. they allowed the venafi api access but still weren't able to figure out the actual issue. I'm not sure why it is closed out stream connection here. 
there is on going issue too, we're frequently getting 500 error message with our venafi site as well.  But from the UI and venaficli with same credentials, I was able to create the cert.

**Comment by arykalin** [comment]
Hi @zerolaser Could you show your venafi provider config, with sensitive adtahidden of course? Did you set zone parameter in provider configuration?

**Comment by zerolaser** [comment]
@arykalin yes. I'm using zone config in the provider configuration.  I parameterized everything.let me know if you need any information.

```
provider "venafi" {
    url          = "${var.venafi_url}"
    tpp_username = "${var.user}"
    tpp_password = "${var.pass}"
    zone = "{var.zone}"
}


resource "venafi_certificate" "webserver" {
    common_name = "${var.cname}"
    key_password = "${var.pk_pass}"
    ecdsa_curve = "p256"

}
```

**Comment by arykalin** [comment]
I see a typo in your provider configuration - zone = "{var.zone}" there should be zone = "${var.zone}"

**Comment by zerolaser** [comment]
@arykalin Thanks. that was a dumb typo. 
you haven't mentioned anything about certificate_dn in the readme file. 
```venafi_certificate.webserver: Creating...
  algorithm:         "" => "RSA"
  certificate:       "" => "<computed>"
  certificate_dn:    "" => "<computed>"
  chain:             "" => "<computed>"
  common_name:       "" => "test3.nikecloud.net"
  csr_pem:           "" => "<computed>"
  ecdsa_curve:       "" => "p256"
  expiration_window: "" => "168"
  key_password:      "<sensitive>" => "<sensitive>"
  private_key_pem:   "" => "<computed>"
  rsa_bits:          "" => "2048"

Error: Error applying plan:

1 error(s) occurred:

* venafi_certificate.webserver: 1 error(s) occurred:

* venafi_certificate.webserver: Unexpected status code on TPP Certificate Request. Status: 400 Bad Request, check Error in response for details.. Body: {"Error":"Need a certificate authority (CADN).\u000d\u000a"}: {"Error":"Need a certificate authority (CADN).\u000d\u000a"}

Terraform does not automatically rollback in the face of errors.
Instead, your Terraform state file has been partially updated with
any resources that successfully completed. Please address the error
above and apply again to incrementally change your infrastructure.```

can you point me where to look for this `certificate_dn` on venafi website ?

**Comment by arykalin** [comment]
certificate_dn will be set automatically to the request ID after request is sent. At this moment we don't use it.

**Comment by zerolaser** [comment]
@arykalin It failed for me throwing that error. what does that mean ?

**Comment by arykalin** [comment]
Oh, I'm sorry, didn't recognize you have this error. Do you have CA templates assigned to TPP policy?
As it said in https://github.com/Venafi/terraform-provider-venafi#requirements-for-usage-with-trust-protection-platform

**Comment by zerolaser** [comment]
@arykalin yes. we assign the TLS template to the internal certs and Entrust template to the external certs. How do i get around that ?

**Comment by zerolaser** [comment]
@arykalin any update on this ?

**Comment by arykalin** [comment]
@zerolaser sorry for delay. You mean you have default CA template configured for the policy which you use for terraform? Could you check policy permissions and that you entering right policy name in configuration? Also look into comment here https://github.com/Venafi/terraform-provider-venafi/issues/7#issuecomment-434116378 there is same error.

**Comment by tr1ck3r** [comment]
@zerolaser following up on this to confirm you were able to resolve this by updating settings on the TPP side?  I also wanted to let you know that we just released v0.8.0 which adds support for Terraform v0.12.x.
